src/dev/jest/tsconfig.json

```diff
@@ -0,0 +1,17 @@
+{
+  // Specific changes to TS config needed to run the tests.
+
+  "extends": "../../../tsconfig.json",
+  "compilerOptions": {
+    "module": "commonjs",
+    // In production we run the TS output through Babel or Webpack, so we can
+    // output `esnext` while the tests are ran directly on the TypeScript
+    // output. We therefore need to the target to match what is available when
+    // Jest is running. When upgrading Node.js versions we can therefore likely
+    // update the target to a newer version of EcmaScript.
+    "target": "es2015"
+  },
+  "include": [
```

Comment:
Did some testing, looks like we maybe can do this instead:

```diff
diff --git a/src/dev/jest/tsconfig.json b/src/dev/jest/tsconfig.json
index e19a53a81e..0bfce29894 100644
--- a/src/dev/jest/tsconfig.json
+++ b/src/dev/jest/tsconfig.json
@@ -12,6 +12,6 @@
     "target": "es2015"
   },
   "include": [
-    "../../../packages/kbn-pm/**/*"
+    "../../../**/*.ts"
   ]
 }
diff --git a/tsconfig.json b/tsconfig.json
index afece44dff..76871cfb27 100644
--- a/tsconfig.json
+++ b/tsconfig.json
@@ -43,6 +43,6 @@
   },
 
   "exclude": [
-    "node_modules"
+    "**/node_modules/**"
   ]
 }
```

What do you think?

Reply:
`"../../../**/*.ts"` ++

What is the use case where `node_modules` doesn't work, but `"**/node_modules/**"` works?

Reply:
It looked like it only included the top-level `node_modules`, not _all_ node_modules in the tree (e.g. within `packages`). I didn't look deeper into that though, so might be mistaken.

Reply:
I believe it excludes every `node_modules`. Just to double check I've experimented with _bad_ file inside `src/node_modules` and it worked as expected: file has been skipped. Actually TS excludes `node_modules` by default (http://www.typescriptlang.org/docs/handbook/tsconfig-json.html), but I still prefer to mention that explicitly.

Reply:
:tada: That's great. It was just the TS output that confused me, thanks for checking.